plugin/trino-hive/src/test/java/io/trino/plugin/hive/avro/TestAvroSchemaEvolutionWithPartitioning.java

```diff
@@ -0,0 +1,327 @@
+    private static final String RENAME_COLUMN_SCHEMA_FILE_NAME = "rename_column_schema.avsc";
+    private static final String REMOVE_COLUMN_SCHEMA_FILE_NAME = "remove_column_schema.avsc";
+    private static final String ADD_COLUMN_SCHEMA_FILE_NAME = "add_column_schema.avsc";
+    private static final String CHANGE_COLUMN_TYPE_SCHEMA_FILE_NAME = "change_column_type_schema.avsc";
+    private static final String INCOMPATIBLE_TYPE_SCHEMA_FILE_NAME = "incompatible_type_schema.avsc";
+
+    private HiveMinioDataLake dockerizedDataLake;
+    private String testTableName;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.dockerizedDataLake = closeAfterClass(
+                new HiveMinioDataLake(
+                        BUCKET_NAME,
+                        ImmutableMap.of(
+                                "avro_resources/core-site.xml", "/etc/hadoop/conf/core-site.xml")));
+        this.dockerizedDataLake.start();
+        this.dockerizedDataLake.copyResourcesToDataLake("avro_resources/schema/", "");
+
+        return S3HiveQueryRunner.create(
+                this.dockerizedDataLake.getHiveHadoop().getHiveMetastoreEndpoint(),
+                this.dockerizedDataLake.getMinio().getMinioApiEndpoint(),
+                HiveMinioDataLake.ACCESS_KEY,
+                HiveMinioDataLake.SECRET_KEY,
+                ImmutableMap.<String, String>builder()
+                        // This is required when using MinIO which requires path style access
+                        .put("hive.s3.path-style-access", "true")
+                        .build());
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        this.testTableName = getTestTableName(TABLE_NAME);
+        getQueryRunner()
+                .execute("CREATE SCHEMA " + CATALOG_NAME + "." + SCHEMA_NAME +
+                        " WITH ( location = 's3a://" + BUCKET_NAME + "/" + SCHEMA_NAME + "')");
+    }
+
+    @BeforeMethod
+    public void createAndLoadTable()
+    {
+        computeActual(format(
+                "CREATE TABLE %s.%s (" +
+                        "  dummy_col VARCHAR, " +
+                        "  partition_col VARCHAR " +
+                        ") WITH (" +
+                        "  format='AVRO', " +
+                        "  partitioned_by=ARRAY['partition_col'], " +
+                        "  avro_schema_url='%s'" +
+                        ")",
+                CATALOG_NAME,
+                testTableName,
+                getAvroSchemaLocation(ORIGINAL_SCHEMA_FILE_NAME)));
+        computeActual(format("INSERT INTO %s VALUES ('stringA', 0, 'stringB')", testTableName));
+    }
+
+    @AfterMethod
+    public void dropTestTable()
+    {
+        query(format("DROP TABLE IF EXISTS %s.%s", CATALOG_NAME, testTableName));
+    }
+
+    @Test
+    public void testSelectTable()
+    {
+        query(format("SELECT string_col FROM %s.%s", CATALOG_NAME, testTableName))
```

Comment:
Can we try something like this 
```
        assertQuery(
                "SELECT string_col FROM " + tableName,
                "VALUES ('stringA')");
```

Reply:
And for all the assertions.

Reply:
You mean instead of building results manually comparing with one returned from H2 as a rule for all test ?